uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -177,6 +178,25 @@ define([
     }
 
     /**
+     * Checks if the balance change has changed since this leave request was saved
+     *
+     * @return {Promise}
+     */
+    function checkIfBalanceChangeHasChanged () {
+      if (!vm.isMode('edit') || vm.isRole('staff')) { return; }
+
+      return LeaveRequest.calculateBalanceChange(_.pick(vm.request,
```

Comment:
Please move `LeaveRequest.calculateBalanceChange` to the request instance ([around here](https://github.com/civicrm/civihr/blob/338fd5f1261227b766ef6ac8483289d59121250a/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/instances/leave-request.instance.js#L230)).

The code should be updated to something like `vm.request.calculateBalanceChange().then(...)`.

The leave request popup tab component should also be updated too [around here](https://github.com/civicrm/civihr/blob/338fd5f1261227b766ef6ac8483289d59121250a/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js#L151).

The reason for this is to make it easier to calculate the balance change for the request without the need to know which parameters to pass.

Reply:
Awesome idea.

Reply:
I `grep`ped calculateBalanceChange and it seems the model method was only used in request.controller.js and request-details-tab.component.js - so all good!